flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/TableStoreSource.java

```diff
@@ -120,6 +121,10 @@ public ChangelogMode getChangelogMode() {
         } else if (table instanceof ChangelogWithKeyFileStoreTable) {
```

Comment:
Nit: in the long run, I think we can simplify the configuration to some extent, e.g., depend on `ChangelogProducer`